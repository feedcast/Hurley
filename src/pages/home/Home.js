import React, { Component } from 'react';
import Helmet from 'react-helmet';

import helpers from 'app/scripts/helpers'

import LatestEpisodes from './components/LatestEpisodes'
import CategoriesChannelList from './components/categoriesChannelList'

import 'app/styles/home.sass'

const Home = ({page, per_page, categories, ...props}) => {
  const { lc } = helpers.localize(this);
  const title = "Feedcast | " + lc.home;

  return (
    <div className="feedcast__home">
      <Helmet title={title} meta={[ {property: 'og:title', content: title} ]} />

      <div className="feedcast__section">
        <LatestEpisodes page={ page } per_page={ 12 } episodes={[]} / >
        <CategoriesChannelList categories={categories}/>
      </div>
    </div>
  );
}

export default Home;
