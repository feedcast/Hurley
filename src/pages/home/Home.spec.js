import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';

import Home from './Home';

describe('Home', () => {
  let component, props;

  beforeEach(() => {
    const home = <Home {...props} />;
    component = renderer.create(withStore(home));
  });

  describe('when has no categories', () => {
    const props = { categories: [], episodes: [] };

    it("renders the spinner", () => {
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has categories', () => {
    const props = {
      categories: [
        {
          "uuid": "f577cc20-6f24-0135-26f9-6eaa3e61655f",
          "slug": "dragoes-de-garagem",
          "title": "Dragões de Garagem",
          "description": "O Dragões de Garagem é um asdasda",
          "feed_url": "http://dragoesdegaragem.com/feed/podcast/",
          "image_url": "https://img.feedcast.io/http%3A%2F%2Fdragoesdegaragem.com%2Fwp-content%2Fuploads%2Fpowerpress%2Fitunes3000.jpg/300/300",
          "listed": true,
          "synchronization_status": "success",
          "synchronization_status_message": ""
        },
        {
          "uuid": "700bb410-7c81-0135-c1b9-2ad04fbc52e9",
          "slug": "hp-news",
          "title": "HP News ",
          "description": "O HP News Express é um podcast sucinto",
          "feed_url": "https://www.megafono.io/podcast/hp-news/rss.xml",
          "image_url": "https://img.feedcast.io/https%3A%2F%2Fd17choic6g575e.cloudfront.net%2Fchannel%2Fartwork%2Fc10ebc0f-d60c-47ce-ad2e-fca0a4dce570%2Flogo-hpnews-megafone-1400px.png/300/300",
          "listed": true,
          "synchronization_status": "success",
          "synchronization_status_message": ""
        }
      ], episodes: [] };

    it("renders the page", () => {
      expect(component).toMatchSnapshot();
    })
  });
})
