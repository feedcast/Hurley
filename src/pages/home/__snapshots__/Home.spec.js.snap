// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home when has categories renders the page 1`] = `
<div
  className="feedcast__home"
>
  <div
    className="feedcast__section"
  >
    <div
      className="feedcast__last-episodes"
    >
      <h3>
         Recents 
      </h3>
      <div
        className="feedcast__episodes-list"
      >
        <div
          className="feedcast__load-more"
        >
          
        </div>
      </div>
    </div>
    <div
      className="feedcast__categories-channel-list"
    >
      <h3>
        You might enjoy
      </h3>
    </div>
  </div>
</div>
`;

exports[`Home when has no categories renders the spinner 1`] = `
<div
  className="feedcast__home"
>
  <div
    className="feedcast__section"
  >
    <div
      className="feedcast__last-episodes"
    >
      <h3>
         Recents 
      </h3>
      <div
        className="feedcast__episodes-list"
      >
        <div
          className="feedcast__load-more"
        >
          
        </div>
      </div>
    </div>
    <div
      className="feedcast__categories-channel-list"
    >
      <h3>
        You might enjoy
      </h3>
    </div>
  </div>
</div>
`;
